vpr/src/route/rr_graph2.cpp

```diff
@@ -1082,26 +1082,88 @@ static void load_block_rr_indices(const DeviceGrid& grid,
+                 * Deposit all the sides
+                 */
+                if (true == wanted_sides.empty()) {
+                    for (e_side side : {TOP, BOTTOM, LEFT, RIGHT}) {
+                        wanted_sides.push_back(side);
+                    }
+                }
+
                 //Assign indices for IPINs and OPINs at all offsets from root
                 for (int ipin = 0; ipin < type->num_pins; ++ipin) {
-                    for (int width_offset = 0; width_offset < type->width; ++width_offset) {
-                        int x_tile = x + width_offset;
-                        for (int height_offset = 0; height_offset < type->height; ++height_offset) {
-                            int y_tile = y + height_offset;
-                            for (e_side side : SIDES) {
+                    bool found_ipin_node = false;
+                    for (e_side side : wanted_sides) {
+                        for (int width_offset = 0; width_offset < type->width; ++width_offset) {
+                            int x_tile = x + width_offset;
+                            for (int height_offset = 0; height_offset < type->height; ++height_offset) {
+                                int y_tile = y + height_offset;
                                 if (type->pinloc[width_offset][height_offset][side][ipin]) {
                                     int iclass = type->pin_class[ipin];
                                     auto class_type = type->class_inf[iclass].type;
 
                                     if (class_type == DRIVER) {
                                         indices[OPIN][x_tile][y_tile][side].push_back(*index);
                                         indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        ++(*index);
                                     } else {
                                         VTR_ASSERT(class_type == RECEIVER);
-                                        indices[IPIN][x_tile][y_tile][side].push_back(*index);
                                         indices[OPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        if (false == found_ipin_node) {
+                                            indices[IPIN][x_tile][y_tile][side].push_back(*index);
+                                            ++(*index);
+                                            /* Found the input pin, no more indexing is required */
+                                            found_ipin_node = true;
+                                        } else {
+                                            VTR_ASSERT_SAFE(true == found_ipin_node);
+                                            indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        }
                                     }
-                                    ++(*index);
                                 } else {
```

Comment:
This code looks wrong; should put (*index), instead of ++(*index) into this location to say we want to connect to the existing rr-node, rather than making a new one.  But right now the code seems to instead just not put any rr-node on that side, which I think is later causing edges to not be connected (based on your graphics and QoR).